### PR TITLE
[BE] github actions migration

### PR DESCRIPTION
**Pull request by aclegg3**
## Motivation and Context

Working branch to test migration to github actions. Do not merge.

## How Has This Been Tested

CI

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by 0mdc** [comment]
@aclegg3 EGL initialization on the Actions runner now works correctly. You may still have issues with the CUDA build - let me know if that happens.

File: .github/actions/install_ubuntu_deps/action.yml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    # we'll use later for the JS build only.
+    run: |
+      echo $(date +%F) > ./date
+      echo $(git ls-remote https://github.com/facebookresearch/habitat-lab.git HEAD | awk '{ print $1}') > ./hablab_sha
+      cat ./hablab_sha
+      wget https://cmake.org/files/v3.12/cmake-3.12.4-Linux-x86_64.sh
+      sudo mkdir /opt/cmake312
+      sudo sh ./cmake-3.12.4-Linux-x86_64.sh --prefix=/opt/cmake312 --skip-license
+      sudo ln -s /opt/cmake312/bin/cmake /usr/local/bin/cmake
+      sudo ln -s /opt/cmake312/bin/ctest /usr/local/bin/ctest
```
Comment (0mdc) [inline .github/actions/install_ubuntu_deps/action.yml:16]: I would simplify this with `sudo apt install -y cmake`.

File: .github/actions/install_ubuntu_deps/action.yml [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  # - name: Install cuda
+  #   run: |-
+  #     echo "Install cuda"
+  #     wget https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2204/x86_64/cuda-keyring_1.1-1_all.deb
+  #     sudo dpkg -i cuda-keyring_1.1-1_all.deb
+  #     sudo apt-get update
+  #     sudo apt-get -y install cuda-toolkit-12-3
+  #     touch ~/cuda_installed
+  #     if command -v nvidia-smi; then nvidia-smi; fi
+  #   shell: bash
```
Comment (0mdc) [inline .github/actions/install_ubuntu_deps/action.yml:52]: We can probably remove this section.

File: .github/workflows/install_and_test.yml [hunk 3]
```diff
@@ -0,0 +1,432 @@
+name: Install and test
+on:
+  pull_request: {}
+  push:
+    branches:
+    - main
+    tags: [ "v*" ]
+  schedule:
+    - cron: "0 5 * * *"
+    #this is 9PM PST
```
Comment (0mdc) [inline .github/workflows/install_and_test.yml:10]: 👍

File: tests/test_configs.py [hunk 4]
```diff
@@ -27,8 +27,9 @@ def test_core_configuration():
     assert config.has_value("test")
     assert config.get("test") == "test statement"
 
-    config.remove("test")
-    assert not config.has_value("test")
+    # TODO: this line segfaults on some platforms, investigate
+    # config.remove("test")
+    # assert not config.has_value("test")
```
Comment (0mdc) [inline tests/test_configs.py:32]: This is pretty strange. Do you have an idea of what could be going on here?

**Review by 0mdc — APPROVED** [review]
LGTM! Good job figuring this out!

File: tests/test_configs.py (hunk 4 above)
Comment (aclegg3) [reply tests/test_configs.py:32]: I have this issue on my local platform (Fedora 40) also. I spoke to John about it. His guess is that the more recent compilers may be treating some C++ code differently in the Configuration string manipulation system. E.g. I have gcc 14 here. He's looking into it, but for now we'll disable this one line. It really only hits when trying to remove a string key from a Configuration which I don't think we ever do anywhere downstream.

File: .github/actions/install_ubuntu_deps/action.yml (hunk 1 above)
Comment (aclegg3) [reply .github/actions/install_ubuntu_deps/action.yml:16]: Can try this and see if it works. :+1:

File: tests/test_configs.py (hunk 4 above)
Comment (jturner65) [reply tests/test_configs.py:32]: (Putting this here for posterity) Yeah, specifically removing a string key to a string field in a raw configuration only.

**Pull request merged**